### PR TITLE
ocamlPackages.memprof-limits: init at 0.2.1

### PR DESCRIPTION
**Pull request by Alizter**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/ocaml-modules/memprof-limits/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    hash = "sha256-Pmuln5TihPoPZuehZlqPfERif6lf7O+0454kW9y3aKc=";
+  };
+
+  minimalOCamlVersion = "4.12";
+
+  meta = with lib; {
+    homepage = "https://ocaml.org/p/memprof-limits/latest";
+    description =
+      "Memory limits, allocation limits, and thread cancellation for OCaml";
+    liscense = licenses.lgpl3;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/memprof-limits/default.nix:20]:
```suggestion
    license = licenses.lgpl3;
```

**Comment by Alizter** [comment]
Yes, it seems we need to restrict < 5.0.

**Comment by Alizter** [comment]
@vbgl I've added a check. Do you think it would be better if we introduced maximalOCamlVersion in buildDunePackage?

**Comment by vbgl** [comment]
> Do you think it would be better if we introduced maximalOCamlVersion in buildDunePackage?

It sounds like a good idea. In particular, this would help avoiding silly mistakes such as swapping the arguments to `versionAtLeast`.

File: pkgs/development/ocaml-modules/memprof-limits/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildDunePackage, fetchFromGitLab, ocaml }:
+
+if (lib.versionAtLeast "5.0.0" ocaml.version) then
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/memprof-limits/default.nix:3]:
```suggestion
if (lib.versionAtLeast ocaml.version "5.0.0") then
```

File: pkgs/build-support/ocaml/dune.nix [hunk 3]
```diff
@@ -8,7 +8,8 @@ let Dune =
 ; in
 
 if (args ? minimumOCamlVersion && lib.versionOlder ocaml.version args.minimumOCamlVersion) ||
-   (args ? minimalOCamlVersion && lib.versionOlder ocaml.version args.minimalOCamlVersion)
+   (args ? minimalOCamlVersion && lib.versionOlder ocaml.version args.minimalOCamlVersion) ||
```
Comment (Alizter) [inline pkgs/build-support/ocaml/dune.nix:11]: Is it correct that our minimal ocaml version bound is strict?

**Comment by vbgl** [comment]
Please leave the refactoring of `buildDunePackage` as a separate PR.

**Comment by Alizter** [comment]
@vbgl OK, I'll put that in a separate PR then.

**Comment by vbgl** [comment]
While you’re at it, please double-check what happens to `minimalOCamlVersion` and the like (i.e., arguments that are intended for dune wrapper and not for the underlying `mkDerivation`).

**Comment by Alizter** [comment]
I've dropped the commit touching buildDunePackage (I'll handle that in another PR) and updated the version check. I am using versionOlder and the arguments should be in the correct order this time.

File: pkgs/development/ocaml-modules/memprof-limits/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ lib, buildDunePackage, fetchFromGitLab, ocaml }:
+
+if (lib.versionOlder ocaml.version "5.0.0") then
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/memprof-limits/default.nix:3]:
```suggestion
if !(lib.versionOlder ocaml.version "5.0.0") then
```

**Pull request merged**